src/dioptra/restapi/app.py

```diff
@@ -113,9 +117,8 @@ def health():
         log = LOGGER.new(request_id=str(uuid.uuid4()))  # noqa: F841
         return jsonify("healthy")
 
-    if not inject_dependencies:
-        return app
-
-    FlaskInjector(app=app, modules=modules)
+    if inject_dependencies:
+        injector = Injector(modules)
+        setup_injection(api, injector)
 
-    return app
+    return app, api
```

Comment:
What's the reasoning for changing the return signature of `create_app()` to return the flask-restx `Api` object in addition to the `Flask` app object? The `Api` object is registered with the Flask app and the user shouldn't need to access or modify it directly.

Is there some sort of scoping issue that emerged when stripping Flask-Injector down to setup_injection?

Reply:
The only reason is the unit tests which need to set up injection externally to `create_app()` (in the app fixture).  That requires the restx `Api` object.  I didn't find a way to access `api` via `app`, so I had to return both.  You can get app via api (which is why the `setup_injection()` function only needs api), but not vice versa.  (So you could return api only, if you wanted, but that would have been... even weirder.)

Reply:
A possibly more palatable design might be to change the `create_app()` signature to accept an injector as an arg, rather than `inject_dependencies`.  It could default to None.  If None, it would create a default injector, else it would use the one passed in.  Then, injection setup need only occur in `create_app()`, and there would be no need to return the `api` object.  With this design, it would not be possible to create an app without dependency injection at all, as is possible now (or maybe it would be possible to create a trivial injector with no bindings/modules, which can't inject anything?).  But perhaps that is not important?

Reply:
@chisholm After thinking about this some more, I like your second proposal of being able to override the dependency injection by changing `create_app()` to accept an injector as an argument. The `create_app()` function is intended to just be a factory for creating Flask app objects, and it would make sense that any (and all) dependency injection should occur there. Go ahead and update this branch with this idea.

And yes, I think if you pass something like `injector=Injector([])` you will be able to get back an app without any dependency injection, although as you hint at in your last question, it isn't clear to me why you would ever want that. Without the injection, none of the class-based views will work, and so you'd just have a broken app.

Reply:
Ok, this is done.